central/image/service/service_impl.go

```diff
@@ -510,9 +528,17 @@ func (s *serviceImpl) EnrichLocalImageInternal(ctx context.Context, request *v1.
 
 		// If the image exists and scan / signature verification results do not need an update yet, return it.
 		// Otherwise, reprocess the image.
-		if imgExists && !forceScanUpdate && !forceSigVerificationUpdate {
-			s.informScanWaiter(request.GetRequestId(), existingImg, nil)
-			return internalScanRespFromImage(existingImg), nil
+		if imgExists {
+			if !forceScanUpdate && !forceSigVerificationUpdate {
+				s.informScanWaiter(request.GetRequestId(), existingImg, nil)
+				return internalScanRespFromImage(existingImg), nil
+			}
+
+			log.Debugw("Scan cache miss enriching image with vulnerabilities",
```

Comment:
doesn't `imgExists` mean there was a cache hit?

Reply:
It means an image was found in Central DB, it doesn't mean that the image is 'returned/used' however.

If either `forceScanUpdate` or `forceSigVerificationUpdate` are true then the image, as it exists in DB, is not used - which is the scenario logged here - an image was found but not used.

I'll enhance this log statement further to log the values of `forceScanUpdate` and `forceSigVerificationUpdate` to help answer the 'why' better.

Reply:
https://github.com/stackrox/stackrox/pull/11733/commits/717b8524962d0b894986462827c2a76531fdfdfc